modules/udparsers/m2c00/spcn_trace.py

```diff
@@ -0,0 +1,216 @@
+#!/usr/bin/env python3
+
+"""
+This module contains functions to parse and format SPCN Trace data produced by
+IO drawers.
+
+The SPCN Trace data can be obtained in the following ways:
+    * Using the web interface to the BMC and predecessors of the BMC.
+    * Collecting a Resource Dump.  The SPCN Trace data must be extracted from
+      the dump using a separate Resource Dump tool.
+
+The SPCN Trace is a hex dump of memory containing the following:
+    * ILOG/PTE data
+    * Zero or more trace buffers
+
+This module can also be run standalone as a script.
+"""
+
+import argparse
+import sys
+
+import pel.hexdump as hexdump
+from udparsers.m2c00.ilog import parse_ilog_data
+from udparsers.m2c00.trace import parse_trace_data
+
+
+# Byte values expected at the start of a trace buffer header
+TRACE_BUFFER_HEADER_START = (b'\x02'        # ver field
+                             b'\x20'        # hdr_len field
+                             b'\x01'        # time_flg field
+                             b'\x42')       # endian_flg field ('B')
+
+
+# Valid values for the 'comp' field in a trace buffer header.  This field
+# contains the buffer name.
+TRACE_BUFFER_NAMES = ['IICS', 'IICM', 'POWR', 'FANS', 'INFO', 'ERRL']
+
+
+# Possible hex dump line formats for the SPCN Trace data
+HEX_DUMP_LINE_FORMATS = [
+    # Format for BMC systems
+    'AAAA:  DDDDDDDD DDDDDDDD DDDDDDDD DDDDDDDD  <CCCCCCCCCCCCCCCC>',
+
+    # Format for pre-BMC systems
+    'DD DD DD DD DD DD DD DD DD DD DD DD DD DD DD DD CCCCCCCCCCCCCCCC'
```

Comment:
So I understand why you want to input the format to `parse()` because it is not the format used in `hexdump()`. Is there any need to input the format to `hexdump()`?

Reply:
hexdump() already provides some flexibility in terms of bytes per line and bytes per chunk.  We could probably add more parameters or switch to the line format option in the future if needed.  But I thought it was good for now.

I didn't want to break the current hexdump() interface, and I wasn't sure whether that level of flexibility was required.  When creating a hex dump, a standard format is probably usually fine.  When parsing one, you have to exactly match the format of the lines.

I initially tried to implement parse() using specific parameters to match hexdump(), like bytes_per_line and bytes_per_chunk.  However, there is so much variability and detail to specify.  Does it have an address?  How many bytes?  What characters follow the address before the data bytes? ...  It just got overly complicated.  I found the 'line format' concept to be easier to specify and implement.  At least to me, it provides a quick visual representation of the line, and it allows you to easily put in all the boring details like 'a colon and then 3 spaces follows the address'.

Let me know if you think this is a bad approach or more changes are needed.

Reply:
I discussed this with Matt, and his opinion was to leave hexdump() as is to promote a common hex dump format across plugins.

Reply:
Good point. Sounds good to me.